### PR TITLE
User registration bf

### LINKED ISSUE #11: User Story 10, User Registration
User Story 10, User Registration

As a visitor
When I click on the 'register' link in the nav bar
Then I am on the user registration page ('/register')
And I see a form where I input the following data:
- my name
- my street address
- my city
- my state
- my zip code
- my email address
- my preferred password
- a confirmation field for my password

When I fill in this form completely,
And with a unique email address not already in the system
My details are saved in the database
Then I am logged in as a registered user
I am taken to my profile page ("/profile")
I see a flash message indicating that I am now registered and logged in

### LINKED ISSUE #12: User Story 11, User Registration Missing Details
As a visitor
When I visit the user registration page
And I do not fill in this form completely,
I am returned to the registration page
And I see a flash message indicating that I am missing required fields

### LINKED ISSUE #13: User Story 12, Registration Email must be unique
As a visitor
When I visit the user registration page
If I fill out the registration form
But include an email address already in the system
Then I am returned to the registration page
My details are not saved and I am not logged in
The form is filled in with all previous data except the email field and password fields
I see a flash message telling me the email address is already in use

### PR DESCRIPTION
**Pull request by benfox1216**
**Functionality:**
User Registration

**Testing** (RSpec tests and development environment)
all passing
**Test %: 100

**Related Issues:**
Need to clean up a couple small bugs

Notes:
Closes #11 
Closes #12
Closes #13

### CODE REVIEW

**Comment by danmoran-pro** [comment]
Looks Great thank you!!!

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -1 +1,10 @@
...
+<%= render partial: 'form', locals: {path: '/register',
+                                     method: :post,
+                                     button_text: 'Register',
+                                     name: (@user.name if @user),
+                                     address: (@user.address if @user),
+                                     city: (@user.city if @user),
+                                     state: (@user.state if @user),
+                                     zip_code: (@user.zip_code if @user),
+                                     email_address: (@user.email_address if @user)
+                                    } %>
```
Comment (Yetidancer) [inline app/views/users/new.html.erb:10]: solved!! yahoo!
our if statement only worked with the right paren spacing

File: config/routes.rb [hunk 2]
```diff
@@ -35,6 +35,6 @@
   get "/orders/:id", to: "orders#show"
   
   get "/register", to: "users#new"
-  post "/profile", to: "users#create"
+  post "/register", to: "users#create"
```
Comment (Yetidancer) [inline config/routes.rb:38]: annoying set up but it's how it's supposed to be...

File: spec/features/users/new_spec.rb [hunk 3]
```diff
@@ -17,22 +17,27 @@
...
-      fill_in "Password", with: @password
-      fill_in "Confirm Password", with: @confirm_pw
+      fill_in :name, with: @name
+      fill_in :address, with: @address
+      fill_in :city, with: @city
+      fill_in :state, with: @state
+      fill_in :zip_code, with: @zip
+      fill_in :email_address, with: @email
+      fill_in :password, with: @password
+      fill_in :confirm_password, with: @confirm_pw
```
Comment (Yetidancer) [inline spec/features/users/new_spec.rb:34]: makes sense for the two word fields

**Pull request merged**